include/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.hpp

```diff
@@ -0,0 +1,151 @@
+    /// @param                  [in] aNumericalSolver A Numerical Solver. Defaults to Undefined.
+    /// @param                  [in] aDynamicsArray Optional array of shared dynamics.
+    /// @param                  [in] maximumPropagationDuration Maximum duration for propagation. Defaults to 7.0 days.
+
+    Sequence(
+        const Size& aRepetitionCount = 1,
+        const NumericalSolver& aNumericalSolver = NumericalSolver::Undefined(),
+        const Array<Shared<Dynamics>>& aDynamicsArray = Array<Shared<Dynamics>>::Empty(),
+        const Duration& maximumPropagationDuration = Duration::Days(7.0)
+    );
+
+    /// @brief                  Output stream operator.
+    ///
+    /// @param                  [in] anOutputStream An output stream.
+    /// @param                  [in] aSequence A sequence.
+    /// @return                 An output stream.
+
+    friend std::ostream& operator<<(std::ostream& anOutputStream, const Sequence& aSequence);
+
+    /// @brief                  Get segments.
+    ///
+    /// @return                 Array of segments.
+
+    Array<TrajectorySegment> getSegments() const;
+
+    /// @brief                  Get numerical solver.
+    ///
+    /// @return                 Numerical solver.
+
+    NumericalSolver getNumericalSolver() const;
+
+    /// @brief                  Get dynamics.
+    ///
+    /// @return                 Dynamics.
+
+    Array<Shared<Dynamics>> getDynamics() const;
+
+    /// @brief                  Get maximum propagation duration.
+    ///
+    /// @return                 Maximum propagation duration.
+
+    Duration getMaximumPropagationDuration() const;
+
+    /// @brief                  Add a trajectory segment.
+    ///
+    /// @param                  [in] aTrajectorySegment A trajectory segment.
+
+    void addSegment(const TrajectorySegment& aTrajectorySegment);
+
+    /// @brief                  Add multiple segments.
+    ///
+    /// @param                  [in] aTrajectorySegmentArray An array of trajectory segments.
+
+    void addSegment(const Array<TrajectorySegment>& aTrajectorySegmentArray);
+
+    /// @brief                  Add a coast segment.
+    ///
+    /// @param                  [in] anEventConditionSPtr An event condition.
+
+    void addCoastSegment(const Shared<EventCondition>& anEventConditionSPtr);
+
+    /// @brief                  Add a maneuver segment.
+    ///
+    /// @param                  [in] anEventConditionSPtr An event condition.
+    /// @param                  [in] aThruster A thruster dynamics.
+
+    void addManeuverSegment(const Shared<EventCondition>& anEventConditionSPtr, const Shared<Thruster>& aThruster);
+
+    /// @brief                  Solve the sequence given an initial state.
+    ///
+    /// @param                  [in] aState Initial state for the sequence.
+    /// @return                 A Solution that contains solutions for each segment.
+
+    Solution solve(const State& aState) const;
```

Comment:
Nit: Maybe `run`/`execute`/etc. Coming from the outside, it looked to me as it `solve` was actually finding something out for us. Maybe we can keep `solve` for when we get to differential correction, maneuver planning, etc.